src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1511,6 +1529,56 @@ public bool UsePerfCounters
 
         #region Misc
 
+        internal void InitServiceFile()
+        {
+            var service = Service ?? PostgresEnvironment.Service;
+            var serviceFile = PostgresEnvironment.ServiceFile ?? PostgresEnvironment.ServiceFileDefault;
+            if (string.IsNullOrEmpty(service) || string.IsNullOrEmpty(serviceFile) || !File.Exists(serviceFile))
```

Comment:
If `Service` is set via a connection string or the environment variable, it should not be ignored. For example:

https://github.com/npgsql/npgsql/blob/d413846ad999768b91467219d9f8b5632147e923/src/Npgsql/NpgsqlConnector.Auth.cs#L463-L478

Only defaults allowed to be silently skipped in case of missed files.